### PR TITLE
Add table aws_wellarchitected_lens_review_improvement

### PR DESCRIPTION
**Pull request by ParthaI**
# Integration test logs
<details>
  <summary>Logs</summary>

```
N/A
```
</details>

# Example query results
<details>
  <summary>Results</summary>

```
> select
  lens_alias,
  lens_arn,
  workload_id,
  improvement_plan_url,
  pillar_id,
  question_id,
  question_title
from
  aws_wellarchitected_lens_review_improvement
where
  lens_alias = 'wellarchitected'
  and workload_id = '4fca39b680a31bb118be6bc0d177849d';
+-----------------+---------------------------------------------------+----------------------------------+----------------------------------------------------------------------------------------------------------------+-----------------------+-------------------+------>
| lens_alias      | lens_arn                                          | workload_id                      | improvement_plan_url                                                                                           | pillar_id             | question_id       | quest>
+-----------------+---------------------------------------------------+----------------------------------+----------------------------------------------------------------------------------------------------------------+-----------------------+-------------------+------>
| wellarchitected | arn:aws:wellarchitected::aws:lens/wellarchitected | 4fca39b680a31bb118be6bc0d177849d | https://wa.aws.amazon.com/TypeII/en/wellarchitected/wellarchitected.govern-usage.improvement-plan.en.html      | costOptimization      | govern-usage      | How d>
| wellarchitected | arn:aws:wellarchitected::aws:lens/wellarchitected | 4fca39b680a31bb118be6bc0d177849d | https://wa.aws.amazon.com/TypeII/en/wellarchitected/wellarchitected.protect-data-rest.improvement-plan.en.html | security              | protect-data-rest | How d>
| wellarchitected | arn:aws:wellarchitected::aws:lens/wellarchitected | 4fca39b680a31bb118be6bc0d177849d | https://wa.aws.amazon.com/TypeII/en/wellarchitected/wellarchitected.telemetry.improvement-plan.en.html         | operationalExcellence | telemetry         | How d>
| wellarchitected | arn:aws:wellarchitected::aws:lens/wellarchitected | 4fca39b680a31bb118be6bc0d177849d | https://wa.aws.amazon.com/TypeII/en/wellarchitected/wellarchitected.mit-deploy-risks.improvement-plan.en.html  | operationalExcellence | mit-deploy-risks  | How d>
+-----------------+---------------------------------------------------+----------------------------------+----------------------------------------------------------------------------------------------------------------+-----------------------+-------------------+------>

Time: 6.7s. Rows fetched: 4. Hydrate calls: 0.
> select
  lens_alias,
  lens_arn,
  workload_id,
  improvement_plan_url,
  question_id,
  question_title,
  risk
from
  aws_wellarchitected_lens_review_improvement
where
  lens_alias = 'wellarchitected'
  and workload_id = '4fca39b680a31bb118be6bc0d177849d'
  and risk = 'HIGH';
+-----------------+---------------------------------------------------+----------------------------------+--------------------------------------------------------------------------------------------------------+-------------+-------------------------------------------->
| lens_alias      | lens_arn                                          | workload_id                      | improvement_plan_url                                                                                   | question_id | question_title                             >
+-----------------+---------------------------------------------------+----------------------------------+--------------------------------------------------------------------------------------------------------+-------------+-------------------------------------------->
| wellarchitected | arn:aws:wellarchitected::aws:lens/wellarchitected | 4fca39b680a31bb118be6bc0d177849d | https://wa.aws.amazon.com/TypeII/en/wellarchitected/wellarchitected.telemetry.improvement-plan.en.html | telemetry   | How do you design your workload so that you>
+-----------------+---------------------------------------------------+----------------------------------+--------------------------------------------------------------------------------------------------------+-------------+-------------------------------------------->

Time: 35ms. Rows fetched: 4 (cached). Hydrate calls: 0.
> select
  lens_arn,
  workload_id,
  risk,
  count(risk)
from
  aws_wellarchitected_lens_review_improvement
where
  lens_alias = 'wellarchitected'
  and workload_id = '4fca39b680a31bb118be6bc0d177849d'
group by
  risk,
  lens_arn,
  workload_id;
+---------------------------------------------------+----------------------------------+--------+-------+
| lens_arn                                          | workload_id                      | risk   | count |
+---------------------------------------------------+----------------------------------+--------+-------+
| arn:aws:wellarchitected::aws:lens/wellarchitected | 4fca39b680a31bb118be6bc0d177849d | HIGH   | 1     |
| arn:aws:wellarchitected::aws:lens/wellarchitected | 4fca39b680a31bb118be6bc0d177849d | MEDIUM | 3     |
+---------------------------------------------------+----------------------------------+--------+-------+

Time: 17ms. Rows fetched: 4 (cached). Hydrate calls: 0.
```
</details>

### CODE REVIEW

File: aws/table_aws_wellarchitected_lens_review_improvement.go [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+
+		input := &wellarchitected.ListLensReviewImprovementsInput{
+			WorkloadId: workload.WorkloadId,
+			LensAlias:  aws.String(lensAlias),
+			MaxResults: maxLimit,
+		}
+
+		if d.EqualsQuals["milestone_number"] != nil {
+			input.MilestoneNumber = int32(d.EqualsQuals["milestone_number"].GetInt64Value())
+			plugin.Logger(ctx).Debug("aws_wellarchitected_lens_review_improvement.listWellArchitectedLensReviewImprovements", "milestone_number", input.MilestoneNumber)
```
Comment (misraved) [inline aws/table_aws_wellarchitected_lens_review_improvement.go:164]: Do we get a validation error if the `milestone_number` is more than 100?

https://docs.aws.amazon.com/wellarchitected/latest/APIReference/API_ListLensReviewImprovements.html

If not we should return an error saying that the value should be less than 100?

File: aws/table_aws_wellarchitected_lens_review_improvement.go [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+			Hydrate:       listWellArchitectedLensReviewImprovements,
+			// TODO: Uncomment and remove extra check in
+			// listWellArchitectedLensReviewImprovements function once this works again
+			// IgnoreConfig: &plugin.IgnoreConfig{
+			// 	ShouldIgnoreErrorFunc: shouldIgnoreErrors([]string{"ResourceNotFoundException", "ValidationException"}),
+			// },
+			KeyColumns: []*plugin.KeyColumn{
+				{Name: "workload_id", Require: plugin.Optional},
+				{Name: "lens_alias", Require: plugin.Optional},
+				{Name: "milestone_number", Require: plugin.Optional},
```
Comment (misraved) [inline aws/table_aws_wellarchitected_lens_review_improvement.go:36]: Should we not include `pillar_id` as an optional parameter?

https://docs.aws.amazon.com/wellarchitected/latest/APIReference/API_ListLensReviewImprovements.html

File: docs/tables/aws_wellarchitected_lens_review_improvement.md [hunk 3]
```diff
@@ -0,0 +1,76 @@
+# Table: aws_wellarchitected_lens_review_improvement
+
+The improvement plan items for your lens review.
+
+**Note:**
+- `workload_id` and `lens_alias` optional in the query parameter for filtering out the review improvements with given workload id or lens alias.
```
[the remark this answers is not in the document]
Comment (ParthaI) [reply docs/tables/aws_wellarchitected_lens_review_improvement.md:6]: updated

File: aws/table_aws_wellarchitected_lens_review_improvement.go [hunk 4]
```diff
@@ -0,0 +1,219 @@
...
+				}
+			}
+			plugin.Logger(ctx).Error("aws_wellarchitected_lens_review_improvement.stereamlistLensReviewImprovements", "api_error", err)
+			return nil, err
+		}
+
+		for _, item := range output.ImprovementSummaries {
+			// For Custom Lenses the lens alias is same as lens arn.
+			// https://docs.aws.amazon.com/wellarchitected/latest/APIReference/API_LensSummary.html#wellarchitected-Type-LensSummary-LensAlias
+			if output.LensAlias == nil || *output.LensAlias == "" {
```
[the remark this answers is not in the document]
Comment (ParthaI) [reply aws/table_aws_wellarchitected_lens_review_improvement.go:200]: Removed double check

File: aws/table_aws_wellarchitected_lens_review_improvement.go (hunk 1 above)
Comment (ParthaI) [reply aws/table_aws_wellarchitected_lens_review_improvement.go:164]: Added the condition

File: aws/table_aws_wellarchitected_lens_review_improvement.go (hunk 2 above)
Comment (ParthaI) [reply aws/table_aws_wellarchitected_lens_review_improvement.go:36]: Added pillar_id as an optional quals.

**Review by misraved — APPROVED** [review]

**Pull request merged**